### ISSUE TITLE
TensorBoard 統合による学習メトリクスのリアルタイムモニタリング

### ISSUE THREAD

**Issue by kurorosu** [kurorosu/pochidetection#346]
## Summary

学習中のメトリクス (Loss, mAP, 学習率) を TensorBoard でリアルタイムに可視化できるようにする.

## Background

物体検出の学習は時間がかかるため, 学習中にメトリクスの推移をリアルタイムで確認したい. 現在は学習完了後に `training_report.html` で確認するしかなく, 学習中の判断 (途中停止など) が難しい.

pochitrain (#325) で同機能を導入済みであり, 設計を踏襲する. また, 主要 OSS (Ultralytics YOLO, MMDetection, Detectron2, PyTorch Lightning, HuggingFace Trainer) はすべてログ書き出しのみでサーバー自動起動は行わない方式を採用しており, これに倣う.

## Scope

- `TensorBoardWriter` クラスの新規作成 (`torch.utils.tensorboard.SummaryWriter` のラッパー)
- エポック単位で以下を記録:
  - `Loss/train` — 学習損失
  - `Loss/val` — 検証損失
  - `Metrics/mAP` — Mean Average Precision
  - `Metrics/mAP_50` — mAP@50
  - `Metrics/mAP_75` — mAP@75
  - `LearningRate` — 学習率
- 設定パラメータ `enable_tensorboard: bool = False` の追加 (スキーマ, TypedDict, ベース設定)
- 学習ループ (`run_training_loop`) への統合
- 学習開始時にログへ起動コマンドを表示 (Ultralytics 方式):
  ```
  TensorBoard: tensorboard --logdir <workspace_path>/tensorboard
  ```
- ログ出力先: `work_dirs/YYYYMMDD_XXX/tensorboard/`
- `tensorboard` 依存の追加 (`pyproject.toml`)
- テストの追加

## Out of Scope

- TensorBoard サーバーの自動起動 (リモート/ヘッドレス環境, ポート競合の問題があるため)
- 画像やモデルグラフの記録 (将来の拡張として検討)
- W&B, MLflow 等の他ツール統合

## Acceptance Criteria

- `enable_tensorboard = True` で学習実行時, `workspace/tensorboard/` に TensorBoard イベントファイルが出力される
- `enable_tensorboard = False` (デフォルト) では TensorBoard 関連の処理が実行されない
- 学習開始時にログへ `tensorboard --logdir` コマンドが表示される
- 記録項目: train_loss, val_loss, mAP, mAP@50, mAP@75, LR (エポック単位)
- 学習終了時に `SummaryWriter` が `close()` される
- `uv run pytest` が通る
- `uv run pre-commit run --all-files` が通る

## Additional Context

- pochitrain PR #325 で同機能を導入済み. 設計を踏襲する
- 主要 OSS の TensorBoard 統合方式:

| フレームワーク | サーバー自動起動 | ユーザーへの通知 |
|---|---|---|
| Ultralytics YOLO | しない | ログにコマンド表示 |
| MMDetection | しない | ドキュメントで案内 |
| Detectron2 | しない | ドキュメントで案内 |
| PyTorch Lightning | しない | ドキュメントで案内 |
| HuggingFace Trainer | しない | ドキュメントで案内 |

## Labels

`enhancement`

## Branch

`feat/tensorboard-integration`
